### PR TITLE
[DEV-2413] adapt mails

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
- Add migration
- Wait for DEV-2459 and change url

File: src/subdomains/supporting/payment/services/transaction.service.ts [hunk 1]
```diff
@@ -61,4 +75,36 @@ export class TransactionService {
       },
     });
   }
+
+  private async txConfirmedMail(entity: Transaction): Promise<void> {
+    try {
+      if (entity.mailTarget?.userData.mail && !DisabledProcess(Process.TX_MAIL)) {
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/services/transaction.service.ts:81]: Maybe `sendTxAssignedMail` would be a better name? Confirmed sounds like it's already finished 😉

File: src/subdomains/supporting/payment/entities/transaction.entity.ts [hunk 2]
```diff
@@ -67,4 +78,29 @@ export class Transaction extends IEntity {
 
   @ManyToOne(() => User, (user) => user.transactions, { nullable: true, eager: true })
   user: User;
+
+  // --- ENTITY METHODS --- //
+
+  confirmSentMail(): UpdateResult<Transaction> {
+    const update: Partial<BuyCrypto> = {
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/entities/transaction.entity.ts:85]: `mailSent`?

File: src/subdomains/supporting/payment/entities/transaction.entity.ts [hunk 3]
```diff
@@ -67,4 +78,29 @@ export class Transaction extends IEntity {
...
+    return [this.id, update];
+  }
+
+  get url(): string {
+    return `${Config.frontend.services}/tx/${this.id}`;
+  }
+
+  get mailTarget(): BuyCrypto | BuyFiat | undefined {
+    return this.buyCrypto ?? this.buyFiat ?? undefined;
+  }
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/entities/transaction.entity.ts:101]: `target` is a bit weird ... but I don't have a better name yet 😆 🤔

**Comment by Yannick1712** [comment]
- add buyCrypto/bankTx missing Route Mail

File: src/subdomains/supporting/payment/entities/transaction.entity.ts (hunk 3 above)
Comment (davidleomay) [reply src/subdomains/supporting/payment/entities/transaction.entity.ts:103]: Use `targetEntity`?

File: src/subdomains/supporting/payment/entities/transaction.entity.ts [hunk 4]
```diff
@@ -71,6 +81,31 @@ export class Transaction extends IEntity {
...
+    };
+
+    Object.assign(this, update);
+
+    return [this.id, update];
+  }
+
+  get url(): string {
+    return `${Config.frontend.services}/tx/${this.id}`;
+  }
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/entities/transaction.entity.ts:99]: UID?

File: src/subdomains/supporting/notification/factories/mail.factory.ts [hunk 5]
```diff
@@ -32,6 +32,7 @@ export enum MailTranslationKey {
   ACCOUNT_MERGE_CHANGED_MAIL = 'mail.account_merge.changed_mail',
   LIMIT_REQUEST = 'mail.limit_request',
   BLACK_SQUAD = 'mail.black_squad',
+  UNVERIFIED_FIAT_INPUT = 'mail.payment.fiat_input.unverified',
 }
```
Comment (davidleomay) [inline src/subdomains/supporting/notification/factories/mail.factory.ts:36]: unassigned?

File: src/subdomains/supporting/payment/services/transaction.service.ts [hunk 6]
```diff
@@ -21,12 +25,16 @@ export class TransactionService {
...
+    let entity = await this.getTransactionById(id);
     if (!entity) throw new Error('Transaction not found');
 
     Object.assign(entity, dto);
 
-    return this.repo.save(entity);
+    entity = await this.repo.save(entity);
+
+    await this.transactionNotificationService.sendTxAssignedMail(entity);
+
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/services/transaction.service.ts:36]: Only send mail if type was set?

**Pull request merged**